tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -803,3 +804,365 @@ def test_csrlsvqr(self, dtype):
                                    reorder=self.reorder)
         cupy.testing.assert_allclose(x, ref_x, rtol=test_tol,
                                      atol=test_tol)
+
+
+@testing.with_requires('scipy>=1.4')
+@testing.gpu
+# tests adapted from scipy's tests of lobpcg
+class TestLOBPCG:
+
+    def _elasticRod(self, n, xp):
+        """Build the matrices for the generalized eigenvalue problem of the
+        fixed-free elastic rod vibration model.
+        """
+        L = 1.0
+        le = L/n
+        rho = 7.85e3
+        S = 1.e-4
+        E = 2.1e11
+        mass = rho*S*le/6.
+        k = E*S/le
+        A = k*(xp.diag(xp.r_[2.*xp.ones(n-1), 1])-xp.diag(xp.ones(n-1), 1) -
+               xp.diag(xp.ones(n-1), -1))
+        B = mass*(xp.diag(xp.r_[4.*xp.ones(n-1), 2])+xp.diag(xp.ones(n-1), 1)
+                  + xp.diag(xp.ones(n-1), -1))
+        return A, B
+
+    def _mikotaPair(self, n, xp):
+        """Build a pair of full diagonal matrices for the generalized eigenvalue
+        problem. The Mikota pair acts as a nice test since the eigenvalues are
+        the squares of the integers n, n=1,2,...
+        """
+        x = xp.arange(1, n+1)
+        B = xp.diag(1./x)
+        y = xp.arange(n-1, 0, -1)
+        z = xp.arange(2*n-1, 0, -2)
+        A = xp.diag(z)-xp.diag(y, -1)-xp.diag(y, 1)
+        return A, B
+
+    def _compare_solutions(self, A, B, m, xp, sp):
+        """Check eig vs. lobpcg consistency.
+        """
+        n = A.shape[0]
+        numpy.random.seed(0)  # seeding is different in numpy and cupy!
+        V = numpy.random.rand(n, m)
+        X = scipy.linalg.orth(V)
+        eigvals, _ = sp.linalg.lobpcg(A, xp.asarray(X), B=B, tol=1e-5,
+                                      maxiter=30, largest=False)
+        eigvals.sort()
+        # converting to numpy below as there is no cupy general eigen value
+        # in cupy at the moment
+        w, _ = scipy.linalg.eig(cupy.asnumpy(A), b=cupy.asnumpy(B))
+        w.sort()
+        cupy.testing.assert_array_almost_equal(
+            w[:int(m/2)], cupy.asnumpy(eigvals[:int(m/2)]), decimal=2)
```

Comment:
We don't need to check eig v.s. lobpcg. It's OK to just compare scipy's output and cupyx.scipy's output.